android-components/components/concept/engine/src/main/java/mozilla/components/concept/engine/EngineSession.kt

```diff
@@ -602,6 +607,26 @@ abstract class EngineSession(
         NO_DETECTED,
     }
 
+    /**
+     * Represents a shopping product status of currently displayed page.
+     */
+    enum class ProductUrlStatus {
```

Comment:
Note: At the moment I am not sure if I am being overly dramatic here with the enum, but there probably will be an option to disable the feature down the road, so I decided to add this just in case. It's easily removable ;)

Reply:
Changed this to Boolean. I think the status of the feature and the status of the page are separate